resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -419,22 +419,17 @@ private[spark] class MesosClusterScheduler(
 
   private def isContainerLocalAppJar(desc: MesosDriverDescription): Boolean = {
     val isLocalJar = desc.jarUrl.startsWith("local://")
-    val isContainerLocal = desc.conf.getOption("spark.mesos.appJar.local.resolution.mode").exists {
+    val isContainerLocal = desc.conf.get(config.APP_JAR_LOCAL_RESOLUTION_MODE) match {
       case "container" => true
       case "host" => false
-      case other =>
-        logWarning(s"Unknown spark.mesos.appJar.local.resolution.mode $other, using host.")
-        false
-      }
+    }
     isLocalJar && isContainerLocal
   }
 
   private def getDriverUris(desc: MesosDriverDescription): List[CommandInfo.URI] = {
-    val confUris = List(conf.getOption("spark.mesos.uris"),
-      desc.conf.getOption("spark.mesos.uris"),
-      Some(desc.conf.get(SUBMIT_PYTHON_FILES).mkString(","))).flatMap(
-      _.map(_.split(",").map(_.trim))
-    ).flatten
+    val confUris = (conf.get(config.URIS_TO_DOWNLOAD) ++
+      desc.conf.get(config.URIS_TO_DOWNLOAD) ++
+      desc.conf.get(SUBMIT_PYTHON_FILES)).toList
```

Comment:
I'm not sure this is exactly the same? `_.split(",").map(_.trim)` - before `URIS_TO_DOWNLOAD` should be a comma separated list (probably not intentional, but it works) and `URIS_TO_DOWNLOAD` & `SUBMIT_PYTHON_FILES` could have whitespaces

Reply:
Element values will be trimmed as same as it was - ConfigBuilder handles it:

https://github.com/apache/spark/blob/dc46fb77ba70709374b4e5feec60e59e33a1ea52/core/src/main/scala/org/apache/spark/internal/config/ConfigBuilder.scala#L123-L126

https://github.com/apache/spark/blob/dc46fb77ba70709374b4e5feec60e59e33a1ea52/core/src/main/scala/org/apache/spark/internal/config/ConfigBuilder.scala#L48-L50

https://github.com/apache/spark/blob/dc46fb77ba70709374b4e5feec60e59e33a1ea52/core/src/main/scala/org/apache/spark/util/Utils.scala#L2619-L2621